File: app/src/main/java/com/greenbox/coyni/adapters/BaseRecyclerViewAdapter.java
package com.greenbox.coyni.adapters;

import androidx.recyclerview.widget.RecyclerView;

public abstract class BaseRecyclerViewAdapter<T extends RecyclerView.ViewHolder>
        extends RecyclerView.Adapter<T> {

    public abstract void setOnItemClickListener(OnItemClickListener listener);
}

File: app/src/main/java/com/coyni/pos/app/baseclass/BaseRecyclerViewAdapter.kt
package com.coyni.pos.app.baseclass

import androidx.recyclerview.widget.RecyclerView

abstract class BaseRecyclerViewAdapter<T : RecyclerView.ViewHolder> :
    RecyclerView.Adapter<T>() {
    abstract fun setOnItemClickListener(onItemClickListener: OnItemClickListener)
}